File: Preprocess/ensemble.R
ens<-function(method,datalist){
  len = dim(datalist)	
  ans = 0
  for(i in 1:len[1]){
    if(method == 'average'){
      ans = rowMeans(datalist)
    }
    else if(method == 'minimum'){
      ans[i] = min(datalist[i,])
    }
    else if(method == 'maximum'){
      ans[i] = max(datalist[i,])
    }
    else if(method == 'median'){
      ans[i] = median(as.numeric(datalist[i,]))
    }
    else if(method == 'trimmed mean'){
      ans[i] = (sum(datalist[i,])-min(datalist[i,])-max(datalist[i,]))/(len[2]-2)
    }
    else if(method == 'product'){
      ans[i] = 1
      for(j in 1:len[2]){
        ans[i] = ans[i] * datalist[i,j]
      }
    }
  }
  return(ans)
}

File: source_code/ricky/ensemble.py
def tolist(list,label,data):
	list[label] = data
	return list
def ensemble(list,method):
	if method == "average":
		return list.mean(axis=1)
	elif method == "minimum":
		return list.min(axis=1)
	elif method == "maximum":
		return list.max(axis=1)
	elif method == "median":
		return list.median(axis=1)
	elif method == "trimmed mean":
		sum = list.sum(axis=1)
		max = list.max(axis=1)
		min = list.min(axis=1)
		final = (sum-max-min)/(len(list.columns)-2)
		return final
	elif method == "product":
		return list.product(axis=1)